### DISCUSSION TITLE
How to process a ctg chunk without readLAS inside a function?

### DISCUSSION THREAD

**Discussion by norrila** [r-lidar/lidR#724] [General]
Hi,

I am trying to process 60 las tiles where I need to find the mean ground and local maximum of a bunch of plots where my plot centers are location of telecommunication towers. 

I looked at the examples here https://cran.r-project.org/web/packages/lidR/vignettes/lidR-LAScatalog-class.html and am able to use the following code to clip out my plots but I was hoping to be able to find the median of ground and local maximum after the `clip_roi ` is called but before the `writeLAS` operation.

```r
ctg <- readLAScatalog("path/to/las/files/")
shp <- sf::st_read("plot_center.shp")
opt_output_files(ctg) <- "folder/where/to/store/outputs/plot_{XCENTER}_{YCENTER}_{PLOTID}"
plots <- clip_roi(ctg, plot_centers, radius = 11.2)
```
**Is it possible to do what I need without a function?**

If I need to make my own function, do I need to read in each `chunk `(ie. original las tile) to process the data or can I complete the `clip_roi` without defining `opt_output_files `and process just the clip_roi points?

The example code I am looking at from the vignettes with some comments/questions of what I want to do:

```r
#  test function
routine <- function(chunk)
{
  las <- readLAS(chunk) # do i need to read original las file or can I use points output from `clip_roi`
  if (is.empty(las)) return(NULL)

  # get mean of ground points for small buffer around point
  # find local maximum
  # write a new shp file (is this possible inside this function?)
}
options <- list(automerge = TRUE) # merges all chunks as single las object
out <- catalog_apply(ctg, routine, .options = options) # write out single las file
```

**Comment by Jean-Romain** [comment]
Sorry, I did not really understand the question. Why do you want to process the collection of files without reading the file? How do you think you can find the average of the point without loading the point? If I understood properly, you want to extract some plots and extract some metrics simultaneously? Is that correct?
